File: src/main/java/net/dblsaiko/hctm/net/Packets.java
package net.dblsaiko.hctm.net;

import net.dblsaiko.hctm.init.NetworkRegistry;
import net.dblsaiko.hctm.init.net.ClientboundMsgDef;
import net.dblsaiko.hctm.init.net.ServerboundMsgDef;

import static net.dblsaiko.hctm.HctmBase.MOD_ID;

public class Packets {
    private final NetworkRegistry reg = new NetworkRegistry(MOD_ID);

    public final ServerboundMsgDef<DebugNetRequest> debugNetRequest = this.reg.registerServerbound("debug_net_request", DebugNetRequest.CODEC);
    public final ClientboundMsgDef<DebugNetResponse> debugNetResponse = this.reg.registerClientbound("debug_net_response", DebugNetResponse.CODEC);

    public void registerClientHandlers() {
        this.reg.registerClientHandlers();
    }

    public void registerServerHandlers() {
        this.reg.registerServerHandlers();
    }
}

File: src/main/kotlin/common/init/Packets.kt
package net.dblsaiko.hctm.common.init

import net.dblsaiko.hctm.MOD_ID
import net.dblsaiko.hctm.client.packet.onDebugNetUpdateResponse
import net.dblsaiko.hctm.common.packet.onDebugNetUpdateRequest
import net.fabricmc.fabric.api.client.networking.v1.ClientPlayNetworking
import net.fabricmc.fabric.api.networking.v1.ServerPlayNetworking
import net.minecraft.util.Identifier

object Packets {

    object Client {

        val DEBUG_NET_RESPONSE = Identifier(MOD_ID, "debug_net_recv")

        fun register() {
            ClientPlayNetworking.registerGlobalReceiver(DEBUG_NET_RESPONSE, ::onDebugNetUpdateResponse)
        }

    }

    object Server {

        val DEBUG_NET_REQUEST = Identifier(MOD_ID, "debug_net_req")

        fun register() {
            ServerPlayNetworking.registerGlobalReceiver(DEBUG_NET_REQUEST, ::onDebugNetUpdateRequest)
        }

    }

}